docs/oauth.md

```diff
@@ -199,60 +249,92 @@ This set of OAuth Config capabilities is not shared with the regular connector c
 
-![Image](../assets/oauth-desktop-dialog.png)
+If the authorization server supports dynamic port ranges for native applications then that should be used. This is the default behavior for Tableau OAuth clients. For example `http://localhost:*/Callback`.
 
-By clicking the sign-in button, you will be directed to your OAuth Provider's sign in Url in a default browser on your machine where you can input your credentials, Tableau will get the required attributes back and upon complete you will see this screen:
+If the authorization server does not support this, then enable OAUTH_CAP_FIXED_PORT_IN_CALLBACK_URL, and specify a set of `<redirectUrisDesktop>` in the OAuth config. The redirectUrisDesktop must be of the form `http://localhost:[portnumber]/Callback`. If you need to use loopback address, you can enable OAUTH_CAP_SUPPORTS_HTTP_SCHEME_LOOPBACK_REDIRECT_URLS, but the `<redirectUrisDesktop>` should still specify localhost. The callback at runtime will be rewritten to something like http://127.0.0.1:55555/Callback.
 
-![Image](../assets/oauth-desktop-complete.png)
+For more information on see [RFC 8252: Loopback Interface Redirection](https://datatracker.ietf.org/doc/html/rfc8252#section-7.3).
+
+## External/Custom OAuth Configs on Desktop
+*\*Available starting in Tableau 2023.1*
+
+The end user or admin can specify custom OAuth configs for desktop applications, independent of the plugin connector. These should be installed in `My Tableau Repository/OAuthConfigs` or `My Tableau Prep Repository/OAuthConfigs`. They must define the new field `oauthConfigId`. The user will be prompted to select from available configs when creating the connection.
 
 # OAuth on Tableau Server & Tableau Online
 
-## Server-Wide OAuth Clients
+## Server Level OAuth Clients
+**This is an older approach that may not work for plugins with with multiple OAuth configs. In most cases [Site Level OAuth Clients](#site-level-oauth-clients) are preferred.**
 
-Config your OAuth client on your server: This would be the first step you need to perform for enabling OAuth on Tableau Server, this will setup OAuth Client information to be used on Tableau Server, for example:
+You need to configure the OAuth client on server for all connectors using saved credentials (aka Custom OAuth), which includes all plugins. Execute the following command, substituting the values for your OAuth client.
 ```
-tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"[your_dbclass]\", \"oauth.config.client_id\":\"[your_client_id]\", \"oauth.config.client_secret\":\"[your_client_secret]\", \"oauth.config.redirect_uri\":\"[your_redirect_url]\"}]" --force-keys
+tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"[dbclass]\", \"oauth.config.client_id\":\"[client_id]\", \"oauth.config.client_secret\":\"[client_secret]\", \"oauth.config.redirect_uri\":\"[redirect_url]\"}]" --force-keys
```

Comment:
So another hack.. for plugin connector with multiple idps, the dbclass field need to be defined as "<dbclass>:<oauthConfigId>" not just "dbClass". Cuz currently we don't have another field to differentiate the CI/CS for different idps. 

Code reference: https://sourcegraph.prod.tableautools.com/teams/near/-/blob/modules/connectors/tabconnectorsbase/main/db/PluginOAuthProvider.cpp?L74

Reply:
Okay.